src/haptics.cpp

```diff
@@ -2,27 +2,36 @@
 
 void
 init_haptics() {
-    // Initialize Motor Pins
-    for (int i = 0; i < N_MOTORS; i++) {
-        // Setup EN pin and set to low
-        pinMode(MOTOR_PINS[i] + EN_OFFSET, OUTPUT);
-        digitalWrite(MOTOR_PINS[i] + EN_OFFSET, LOW);
-        pinMode(MOTOR_PINS[i], OUTPUT);
-    }
+  // Initialize Motor Pins
+  for (int i = 0; i < N_MOTORS; i++) {
+    // Setup EN pin and set to low
+    pinMode(MOTOR_PINS[i] + EN_OFFSET, OUTPUT);
+    digitalWrite(MOTOR_PINS[i] + EN_OFFSET, LOW);
+    pinMode(MOTOR_PINS[i], OUTPUT);
+  }
+}
+
+void actuate_motor(int motor_pin, int duration, int frequency, double percent_motor) {
+  if (percent_motor > 0.85) {
+    Serial.println("Motor percentage above 85%. Using 85%");
+    percent_motor = 0.85;
+  }
+  for (int j = 0; j < duration; j++) {  // do J_MAX * TODAL_DELAY = 16 * 6 = 96ms of vibration
+    analogWrite(MOTOR_PINS[motor_pin], 512 + percent_motor * 512);
+    delay(frequency);
+    analogWrite(MOTOR_PINS[motor_pin], 512 - percent_motor * 512);
+    delay(frequency);
+  }
+  digitalWrite(MOTOR_PINS[motor_pin] + EN_OFFSET, LOW);  // disable
 }
 
 /* Test & debug functions */
 void
 haptics_test() {
-    for (int i = 0; i < N_MOTORS; i++) {
-        digitalWrite(MOTOR_PINS[i] + EN_OFFSET, HIGH);  // enable
-        for (int j = 0; j < 64; j++){  // do J_MAX * TODAL_DELAY = 16 * 6 = 96ms of vibe
-            analogWrite(MOTOR_PINS[i], 512 + 0.85 * 512);
-            delay(3);
-            analogWrite(MOTOR_PINS[i], 512 - 0.85 * 512);
-            delay(3);
-        }
-        digitalWrite(MOTOR_PINS[i] + EN_OFFSET, LOW);  // disable
-        delay(50);
-    }
+  for (int i = 0; i < N_MOTORS; i++) {
+    digitalWrite(MOTOR_PINS[i] + EN_OFFSET, HIGH);  // enable
```

Comment:
Suggest moving the enabling of the motor to inside actuate_motor(), since that's where disabling also occurs.